equinix/resource_fabric_service_profile.go

```diff
@@ -22,6 +22,171 @@ import (
+		},
+		"allow_remote_connections": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Setting to allow or prohibit remote connections to the service profile",
+		},
+		"allow_bandwidth_upgrade": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Availability of a bandwidth upgrade. The default is false",
+		},
+		"connection_label": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Custom name for Connection",
+		},
+		"enable_auto_generate_service_key": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Enable auto generate service key",
+		},
+		"bandwidth_alert_threshold": {
+			Type:        schema.TypeFloat,
+			Computed:    true,
+			Description: "Percentage of port bandwidth at which an allocation alert is generated",
+		},
+		"allow_custom_bandwidth": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Setting to enable or disable the ability of the buyer to customize the bandwidth",
+		},
+		"api_config": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Api configuration details",
+			Elem: &schema.Resource{
+				Schema: ApiConfigSch(),
+			},
+		},
+		"authentication_key": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Authentication key details",
+			Elem: &schema.Resource{
+				Schema: AuthenticationKeySch(),
+			},
+		},
+		"link_protocol_config": {
+			Type:        schema.TypeSet,
+			Computed:    true,
+			Description: "Link protocol configuration details",
+			Elem: &schema.Resource{
+				Schema: LinkProtocolConfigSch(),
+			},
+		},
+		"supported_bandwidths": {
+			Type:        schema.TypeList,
+			Computed:    true,
+			Description: "Supported bandwidths",
+			Elem:        &schema.Schema{Type: schema.TypeInt},
+		},
+	}
+}
+
+func ApiConfigSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"api_available": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Setting indicating whether the API is available (true) or not (false)",
+		},
+		"equinix_managed_vlan": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Setting indicating that the VLAN is managed by Equinix (true) or not (false)",
+		},
+		"allow_over_subscription": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Setting showing that oversubscription support is available (true) or not (false). The default is false",
+		},
+		"over_subscription_limit": {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "A cap on over subscription",
+		},
+		"bandwidth_from_api": {
+			Type:        schema.TypeBool,
+			Computed:    true,
+			Description: "Bandwidth from api",
+		},
+		"integration_id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Integration id",
```

Comment:
Curious what this field is used for. The description doesn't provide more information than the name.

Reply:
Same comment about bandwidth_from_api.

Reply:
For this one, and some of the others, the unfortunate answer is that I don't think they actually are useful to the terraform users. They are apart of the Fabric API schema, but are not commonly (if ever) used by a customer to do anything. Difficult to get more information on them but I will double check.

For things like that I would actually like to remove them, but there would be a breaking change as you referenced because we've published v1.0.0 already.

Reply:
Nevermind, I found the details needed to provide a better description. Link here for reference: https://developer.equinix.com/docs?page=/dev-docs/fabric/overview